git-cliff-core/src/commit.rs

```diff
@@ -312,34 +313,49 @@ impl Commit<'_> {
 		for parser in parsers {
 			let mut regex_checks = Vec::new();
 			if let Some(message_regex) = parser.message.as_ref() {
-				regex_checks.push((message_regex, self.message.to_string()));
+				regex_checks.push((message_regex, vec![self.message.to_string()]));
 			}
 			let body = self
 				.conv
 				.as_ref()
 				.and_then(|v| v.body())
 				.map(|v| v.to_string());
 			if let Some(body_regex) = parser.body.as_ref() {
-				regex_checks.push((body_regex, body.clone().unwrap_or_default()));
+				regex_checks
+					.push((body_regex, vec![body.clone().unwrap_or_default()]));
 			}
 			if let (Some(footer_regex), Some(footers)) = (
 				parser.footer.as_ref(),
 				self.conv.as_ref().map(|v| v.footers()),
 			) {
-				regex_checks
-					.extend(footers.iter().map(|f| (footer_regex, f.to_string())));
+				regex_checks.extend(
+					footers.iter().map(|f| (footer_regex, vec![f.to_string()])),
+				);
 			}
 			if let (Some(field_name), Some(pattern_regex)) =
 				(parser.field.as_ref(), parser.pattern.as_ref())
 			{
 				let value = if field_name == "body" {
-					body.clone()
+					vec![body.clone()].into_iter().collect()
 				} else {
 					tera::dotted_pointer(&lookup_context, field_name).and_then(|v| {
-						match &v {
-							Value::String(s) => Some(s.clone()),
+						match v {
+							Value::String(s) => Some(vec![s.clone()]),
 							Value::Number(_) | Value::Bool(_) | Value::Null => {
-								Some(v.to_string())
+								Some(vec![v.to_string()])
+							}
+							Value::Array(arr) => {
+								let mut result = Vec::new();
+								for item in arr {
+									match item {
+										Value::String(s) => result.push(s.clone()),
+										Value::Number(_) |
+										Value::Bool(_) |
+										Value::Null => result.push(item.to_string()),
```

Comment:
This line of the newly implemented code is missing from the code coverage, but since it is theoretically unreachable, I have deemed the lack of coverage here to be acceptable.